### PR TITLE
Implemented #1372: Test concatenation of more types of layers in Flatten...

### PR DESCRIPTION
**Pull request by julianser**
Implemented #1372: Test concatenation of more types of layers in FlattenerLayer. It now includes test for linear, softmax, sigmoid and convolutional layers.

### CODE REVIEW

File: pylearn2/models/tests/test_mlp.py [hunk 1]
```diff
@@ -588,6 +635,199 @@ def test_flattener_layer():
                                    mlp_composite.get_params()[i].eval())
 
 
+def test_flattener_layer():
+    # This function tests that linear, sigmoid and softmax layers are
```
Comment (JesseLivezey) [inline pylearn2/models/tests/test_mlp.py]: I think you may want to turn this into a docstring formatted comment for nose printing. Most of the other tests have a short description in the docstring which helps when things don't work.

File: pylearn2/models/tests/test_mlp.py [hunk 2]
```diff
@@ -435,22 +435,33 @@ def test_get_layer_monitor_channels():
                                       state=None, targets=targets)
 
 
-def test_flattener_layer():
+# Compares two mlp with a single mlp composed of the two mlps
+# and a flattener layer
```
Comment (bouthilx) [inline pylearn2/models/tests/test_mlp.py]: It does what? Which is nested mlp and which is not? 

Please get this comment under function signature and format everything as a docstring like you did [here](https://github.com/lisa-lab/pylearn2/pull/1444/files#diff-102b66694f1d64cec964b556c0a33dd8R638)

File: pylearn2/models/tests/test_mlp.py [hunk 3]
```diff
@@ -588,6 +635,201 @@ def test_flattener_layer():
...
+                                                       first_indep_layer,
+                                                       second_indep_layer,
+                                                       features_in_first_mlp,
+                                                       features_in_second_mlp,
+                                                       targets_in_first_mlp,
+                                                       targets_in_second_mlp)
+
+
+def test_flattener_layer_convolutional_layer():
+    # Testing convolutional layer
```
Comment (bouthilx) [inline pylearn2/models/tests/test_mlp.py]: Docstring. :)

File: pylearn2/models/tests/test_mlp.py [hunk 4]
```diff
@@ -435,22 +435,37 @@ def test_get_layer_monitor_channels():
...
 
-def test_flattener_layer():
-    # To test the FlattenerLayer we create a very simple feed-forward neural
-    # network with two parallel linear layers. We then create two separate
-    # feed-forward neural networks with single linear layers. In principle,
-    # these two models should be identical if we start from the same
-    # parameters. This makes it easy to test that the composite layer works
-    # as expected.
+# Compares two mlp with a single mlp composed of the two mlps
+# and a flattener layer
```
Comment (bouthilx) [inline pylearn2/models/tests/test_mlp.py]: You forgot to remove those comments. 

Thanks for making the docstrings.

**Comment by bouthilx** [comment]
LGTM. I'll merge after last modification. 

@julianser Can you [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) all commits when you are done? Stand alone commits for minor fixes clutter the repo history otherwise.

Thanks! :)

**Comment by bouthilx** [comment]
Thank you Julian!

**Pull request merged**